### PR TITLE
feat: Standardization - Color

### PR DESCRIPTION
**Pull request by lucasaraujo**
Closes #99 

BREAKING CHANGE: Color values (```backgroundColor``` and ```textColor```) must have the ```#``` in front of the hexadecimal and the Alpha component must be at the end. Valid formats: ```#RGB```, ```#RGBA```, ```#RRGGBB```, ```#RRGGBBAA```.

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/beagle/src/main/java/br/com/zup/beagle/utils/ColorUtils.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        return when (hexColor.count()){
+            9-> Color.parseColor(formatHexColorAlpha(hexColor))
+            else -> Color.parseColor(formatHexColor(hexColor))
+        }
+    }
+
+    private fun formatHexColorAlpha(color: String): String {
+        return "^#([0-9A-F])([0-9A-F])([0-9A-F])([0-9A-F])([0-9A-F])([0-9A-F])([0-9A-F])([0-9A-F])$"
+            .toRegex(RegexOption.IGNORE_CASE)
+            .replace(color, "#\$7\$8\$1\$2\$3\$4\$5\$6")
```
Comment (lucasaraujo) [inline android/beagle/src/main/java/br/com/zup/beagle/utils/ColorUtils.kt:33]: The patter could be `"^#([0-9A-F]{6})([0-9A-F]{2})$"` and the replace `"#$2$1"`

File: iOS/Example/BeagleDemo/BeagleDemo/Constants/Constants.swift [hunk 2]
```diff
@@ -59,3 +52,15 @@ extension String {
     static let TEXT_IMAGE_STYLE = "DesignSystem.Text.Image"
     static let TEXT_HELLO_WORD_STYLE = "DesignSystem.Text.helloWord"
 }
+
+extension UIColor {
+    class var demoGreen: UIColor {
+        return UIColor(red: 0.34, green: 0.62, blue: 0.16, alpha: 1)
```
Comment (botbeaglezup) [inline iOS/Example/BeagleDemo/BeagleDemo/Constants/Constants.swift:58]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/1c5f535dcc30da6cd789ac16cbd92858c1ba29dd/iOS/Example/BeagleDemo/BeagleDemo/Constants/Constants.swift#L58"></span>Prefer object literals over image and color inits.<br />
<code>object_literal</code> <code>Constants.swift:58</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Example/BeagleDemo/BeagleDemo/Constants/Constants.swift [hunk 3]
```diff
@@ -59,3 +52,15 @@ extension String {
...
+    }
+    class var demoGray: UIColor {
+        return UIColor(white: 0.5, alpha: 1)
```
Comment (botbeaglezup) [inline iOS/Example/BeagleDemo/BeagleDemo/Constants/Constants.swift:61]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/1c5f535dcc30da6cd789ac16cbd92858c1ba29dd/iOS/Example/BeagleDemo/BeagleDemo/Constants/Constants.swift#L61"></span>Prefer object literals over image and color inits.<br />
<code>object_literal</code> <code>Constants.swift:61</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Example/BeagleDemo/BeagleDemo/Constants/Constants.swift [hunk 4]
```diff
@@ -59,3 +52,15 @@ extension String {
...
+    }
+    class var demoDarkGray: UIColor {
+        return UIColor(white: 0.2, alpha: 1)
```
Comment (botbeaglezup) [inline iOS/Example/BeagleDemo/BeagleDemo/Constants/Constants.swift:64]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/1c5f535dcc30da6cd789ac16cbd92858c1ba29dd/iOS/Example/BeagleDemo/BeagleDemo/Constants/Constants.swift#L64"></span>Prefer object literals over image and color inits.<br />
<code>object_literal</code> <code>Constants.swift:64</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Sources/BeagleUI/Sources/Components/Page+Renderable/Indicator/DefaultPageIndicator+Renderable.swift [hunk 5]
```diff
@@ -51,16 +51,16 @@ class PageIndicatorUIComponent: UIView, PageIndicatorUIView {
     // MARK: - Init
     
     required init(selectedColor: String? = nil, unselectedColor: String? = nil) {
-        if let selected = selectedColor, !selected.isEmpty {
-            self.selectedColor = UIColor(hex: selected)
+        if let selected = selectedColor, let color = UIColor(hex: selected) {
+            self.selectedColor = color
         } else {
-            self.selectedColor = UIColor(hex: "#808080")
+            self.selectedColor = UIColor(white: 0.5, alpha: 1)
```
Comment (botbeaglezup) [inline iOS/Sources/BeagleUI/Sources/Components/Page+Renderable/Indicator/DefaultPageIndicator+Renderable.swift:57]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/1c5f535dcc30da6cd789ac16cbd92858c1ba29dd/iOS/Sources/BeagleUI/Sources/Components/Page+Renderable/Indicator/DefaultPageIndicator+Renderable.swift#L57"></span>Prefer object literals over image and color inits.<br />
<code>object_literal</code> <code>DefaultPageIndicator+Renderable.swift:57</code></td>
    </tr>
  </tbody>
</table>

File: iOS/Sources/BeagleUI/Sources/Components/Page+Renderable/Indicator/DefaultPageIndicator+Renderable.swift [hunk 6]
```diff
@@ -51,16 +51,16 @@ class PageIndicatorUIComponent: UIView, PageIndicatorUIView {
...
         }
         
-        if let unselected = unselectedColor, !unselected.isEmpty {
-            self.unselectedColor = UIColor(hex: unselected)
+        if let unselected = unselectedColor, let color = UIColor(hex: unselected) {
+            self.unselectedColor = color
         } else {
-            self.unselectedColor = UIColor(hex: "#d3d3d3")
+            self.unselectedColor = UIColor(white: 0.8274, alpha: 1)
```
Comment (botbeaglezup) [inline iOS/Sources/BeagleUI/Sources/Components/Page+Renderable/Indicator/DefaultPageIndicator+Renderable.swift:63]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/1c5f535dcc30da6cd789ac16cbd92858c1ba29dd/iOS/Sources/BeagleUI/Sources/Components/Page+Renderable/Indicator/DefaultPageIndicator+Renderable.swift#L63"></span>Prefer object literals over image and color inits.<br />
<code>object_literal</code> <code>DefaultPageIndicator+Renderable.swift:63</code></td>
    </tr>
  </tbody>
</table>

**Pull request merged**